crates/chain/tests/test_tx_graph.rs

```diff
@@ -822,3 +822,95 @@ fn test_additions_last_seen_append() {
         );
     }
 }
+
+/// Inserting anchors without the corresponding transaction
+/// and testing the recoverability of the transaction graph
+#[test]
+fn test_inserting_anchor_without_tx() {
+    let mut graph = TxGraph::<ConfirmationHeightAnchor>::default();
+    let mut graph_recovered = graph.clone();
+    let txs = (0..5).map(common::new_tx).collect::<Vec<_>>();
+    let chain: LocalChain = (0..5)
+        .map(|i| (i, h!(i.to_string())))
+        .collect::<BTreeMap<u32, BlockHash>>()
+        .into();
+    let anchors = (0..5)
+        .map(|i| ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: i,
+                hash: *chain.blocks().get(&i).unwrap(),
+            },
+            confirmation_height: 0,
+        })
+        .collect::<Vec<_>>();
+
+    let mut anchor_additions = Additions::default();
+    for i in 0..3 {
+        let additions = graph.insert_anchor(txs[i].txid(), anchors[i]);
+        anchor_additions.append(additions);
+    }
+
+    // Additions should only include the Anchor for the Tx-Block 0, 1 and 2
+    assert_eq!(
+        anchor_additions,
+        Additions {
+            tx: Default::default(),
+            txout: Default::default(),
+            anchors: BTreeSet::from([
+                (anchors[0], txs[0].txid()),
+                (anchors[1], txs[1].txid()),
+                (anchors[2], txs[2].txid()),
+            ]),
+            last_seen: Default::default(),
+        }
```

Comment:
Nit, for future reference.

```suggestion
        Additions {
            anchors: BTreeSet::from([
                (anchors[0], txs[0].txid()),
                (anchors[1], txs[1].txid()),
                (anchors[2], txs[2].txid()),
            ]),
            ..Default::default()
        }
```